.github/ISSUE_TEMPLATE/release-template.md

```diff
@@ -8,21 +8,49 @@ labels: 'Type: Maintenance'
 
 :warning: DO NOT MERGE (YET) :warning:
 
-[Remaining work for this Milestone](https://github.com/Parsely/wp-parsely/milestone/14)
+[Remaining work for this Milestone](https://github.com/Parsely/wp-parsely/milestone/xx)
 
 PR for tracking changes for the X.Y.Z release. Target release date: DOW DD MMMM YYYY.
 
-- [ ] Merge any outstanding PRs due for this release to `develop`.
+## Before releasing
+
 - [ ] Notify stakeholders of an upcoming release.
-- [ ] Add [PR](https://github.com/Parsely/wp-parsely/pull/XXX) against `develop` which:
-  - Increases version numbers in the README, plugin bootstrap file `wp-parsely.php`, and `package.json`.
-  - Adds or updates the changelog for the release.
-- [ ] Merge this PR.
-- [ ] Create a branch from `trunk` named `release-x.y.z`. Merge `develop` into that branch. Open a PR from `release-x.y.z` to `trunk` named _Release x.y.z_.
-- [ ] Add signed release tag against `trunk` using GitHub's UI.
+- [ ] Merge any outstanding PRs due for this release to `develop`.
+- [ ] Verify that all PRs have a `Changelog` tag.
+
+## Release proccess
+
+**Create the release PR**
+- [ ] Create a PR that merges `develop` into `trunk`, named _Release x.y.z_.
+- [ ] Merge the PR into trunk.
+  > This will create a new release draft. It might take some seconds to appear.
+
+**Promote the release draft to a pre-release**
+- [ ] In the release draft page, check the version numbers and changelog data.
+- [ ] Make any adjustments if needed.
```

Comment:
Can we add a link to https://semver.org/ somewhere here explaining the logic of numbering?

Reply:
Being already lengthy as it is, I'd rather only focus on steps in this document and not mix it with explanations. However, I do see the point of having this info and we should document it somewhere.

However, I can do it if you think it would be better to have the information here.